airflow/cli/cli_parser.py

```diff
@@ -1457,6 +1458,7 @@ def _format_action(self, action: Action):
         return super()._format_action(action)
 
 
+@lru_cache(maxsize=2)
```

Comment:
There is no point in giving this a max size here. There are two possible cache keys: True or False, so the cache is already bounded. But by setting maxsize to 2 the LRU cache decorator will lock and test the cache size every call, and that's just unnecessary.

If you use `maxsize=None` no locking is needed and cache lookups are that little bit faster.

Reply:
Oh thank you, I didn't realize that.

Reply:
For reference, compare the wrapper functions that are returned for the maxsize=None case:

https://github.com/python/cpython/blob/60324d26b58c89d68abb23fb42f1563d395c3910/Lib/functools.py#L548-L561

with the maxsize=2 case:

https://github.com/python/cpython/blob/60324d26b58c89d68abb23fb42f1563d395c3910/Lib/functools.py#L563-L620

Reply:
Very good to know -- that wasn't really made clear from the docs. Doesn't make much difference in this case as it's called in-frequently, but useful.

Reply:
Yeah, that's one of the reasons why 3.9 adds [`functools.cache()`](https://docs.python.org/3/library/functools.html#functools.cache), which is really nothing more than [a proxy for `lru_cache(maxsize=None)`](https://github.com/python/cpython/blob/60324d26b58c89d68abb23fb42f1563d395c3910/Lib/functools.py#L650-L652).

Reply:
Didn't know about functools.cache, nice to know